modules/plantuml/default.nix

```diff
@@ -0,0 +1,36 @@
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
+with lib;
+with builtins; let
+  cfg = config.vim.plantuml;
+in {
+  options.vim = {
+    plantuml = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "enable plantuml previewer";
+      };
+    };
+  };
+
+  config =
+    mkIf cfg.enable
+    {
+      vim.startPlugins = [
+        "plantuml-previewer"
+        pkgs.vimPlugins.plantuml-syntax
+        pkgs.vimPlugins.open-browser
+      ];
+
+      vim.configRC.plantuml = nvim.dag.entryAnywhere ''
+        au filetype plantuml let g:plantuml_previewer#viewer_path = "/tmp/plantuml"
+        au filetype plantuml let g:plantuml_previewer#java_path = "${pkgs.openjdk}/bin/java"
```

Comment:
Can you make two options for setting the packages. eg `vim.plantuml.pkgs.{plantuml,openjdk}` and they can just default to `pkgs.openjdk` etc.

Reply:
yep, done!